### DISCUSSION TITLE
Is the facial recognition system improving itself ?

### DISCUSSION THREAD

**Discussion by 9ycbgf0k8fpg** [photoprism/photoprism#1955] [Q&A]
Hello,

I want to know if face recognition is getting better when I manually tag people on pictures ? Is it learning from my tagging ?

I am asking this because sometimes there are multiple pictures taken in a row (almost identical pictures), and when I tag an unknown face in one picture, it doesn't automatically tag the other pictures as well.

Thanks for your help.

**Comment by graciousgrey** [comment]
PhotoPrism is updating the face clusters in the background when you add names to faces. So, yes it is learning. 

Sometimes you need to label multiple faces or wait a bit. 
Sometimes faces are just not similar enough or the resolution is not sufficient to match them with each other. 
In general facial recognition works best if you have many photos of the same person.

**Comment by 9ycbgf0k8fpg** [comment]
And is it the same with labels ? If a tag a picture of a plate with for instance "salad", will it better recognize salads in the future ?
I am actually in the process of indexing ~25,000 pictures, when I tag someone manually in the already indexed picture, does it learn immediately or do I have to wait for all the pictures to be indexed ? Because it will take around 2 weeks according to my calculations

Thanks a lot for helping me :)

**Reply by graciousgrey** [reply]
For labels there is no learning possible yet.

For the facial recognition learning, you don't need to trigger a re-index this normally happens automatically in the background. But this background worker can't run if your instance is currently busy with indexing other files.
